Teuchos: Linker errors with Sycl backend, intel-oneapi-2025.1.1

## Bug Report
@trilinos/teuchos @trilinos/sacado 

### Description
Nightly builds of Trilinos with the Sycl backend began failing to link following merge of recent updates

**Error output**:
```
[ 28%] Linking CXX executable Sacado_hesopcheck_ADC.exe
../../../teuchos/core/src/libteuchoscore.so.16.2.0: undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > Teuchos::getEnvironmentVariable<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >(std::basic_string_view<char, std::char_traits<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)'
icpx: error: linker command failed with exit code 1 (use -v to see invocation)
gmake[2]: *** [packages/sacado/test/tradoptest/CMakeFiles/Sacado_hesopcheck_ADC.exe.dir/build.make:126: packages/sacado/test/tradoptest/Sacado_hesopcheck_ADC.exe] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:6087: packages/sacado/test/tradoptest/CMakeFiles/Sacado_hesopcheck_ADC.exe.dir/all] Error 2
gmake[1]: *** Waiting for unfinished jobs....
```

**Changes since passing build**:
```
Git (git https://github.com/trilinos/Trilinos.git)

    Removed namespace scoping from std namespace variable declarations ([detail](https://jenkins-son.sandia.gov/view/Kokkos/job/KokkosEco_Trilinos_Blake_OneAPI_2025_1_1_Sycl_PVC_MKL-tpetra_sacado_ifpack2/41/changes#detail))
    Intrepid2: Reduce container overhead for Data, TensorData, VectorData, BasisValues, and TransformedBasisValues (#14546) ([detail](https://jenkins-son.sandia.gov/view/Kokkos/job/KokkosEco_Trilinos_Blake_OneAPI_2025_1_1_Sycl_PVC_MKL-tpetra_sacado_ifpack2/41/changes#detail))
    STK: Snapshot 10-22-25 20:04 from Sierra 5.27.1-647-gd4dfa740 ([detail](https://jenkins-son.sandia.gov/view/Kokkos/job/KokkosEco_Trilinos_Blake_OneAPI_2025_1_1_Sycl_PVC_MKL-tpetra_sacado_ifpack2/41/changes#detail))
    Teuchos: Collect system information ([detail](https://jenkins-son.sandia.gov/view/Kokkos/job/KokkosEco_Trilinos_Blake_OneAPI_2025_1_1_Sycl_PVC_MKL-tpetra_sacado_ifpack2/41/changes#detail))
    SEACAS: Create target mapvar-kd for Performance builds as well ([detail](https://jenkins-son.sandia.gov/view/Kokkos/job/KokkosEco_Trilinos_Blake_OneAPI_2025_1_1_Sycl_PVC_MKL-tpetra_sacado_ifpack2/41/changes#detail))
    Fixing Resize for Sacado SFAD with HIP ([detail](https://jenkins-son.sandia.gov/view/Kokkos/job/KokkosEco_Trilinos_Blake_OneAPI_2025_1_1_Sycl_PVC_MKL-tpetra_sacado_ifpack2/41/changes#detail))
    STK: Snapshot 10-25-25 09:08 from Sierra 5.27.1-812-g02e94e0c ([detail](https://jenkins-son.sandia.gov/view/Kokkos/job/KokkosEco_Trilinos_Blake_OneAPI_2025_1_1_Sycl_PVC_MKL-tpetra_sacado_ifpack2/41/changes#detail))
    Ifpack2: fix string + int concat ([detail](https://jenkins-son.sandia.gov/view/Kokkos/job/KokkosEco_Trilinos_Blake_OneAPI_2025_1_1_Sycl_PVC_MKL-tpetra_sacado_ifpack2/41/changes#detail))
    Tpetra: Eliminate warnings about abs() with clang-19 ([detail](https://jenkins-son.sandia.gov/view/Kokkos/job/KokkosEco_Trilinos_Blake_OneAPI_2025_1_1_Sycl_PVC_MKL-tpetra_sacado_ifpack2/41/changes#detail))
    Teuchos: Eliminate typeid() side-effect warning with clang-19 ([detail](https://jenkins-son.sandia.gov/view/Kokkos/job/KokkosEco_Trilinos_Blake_OneAPI_2025_1_1_Sycl_PVC_MKL-tpetra_sacado_ifpack2/41/changes#detail))
    Teuchos: Eliminate unused var warnings with clang-19 ([detail](https://jenkins-son.sandia.gov/view/Kokkos/job/KokkosEco_Trilinos_Blake_OneAPI_2025_1_1_Sycl_PVC_MKL-tpetra_sacado_ifpack2/41/changes#detail))
    Teuchos systemInformation: Capture cerr for module list ([detail](https://jenkins-son.sandia.gov/view/Kokkos/job/KokkosEco_Trilinos_Blake_OneAPI_2025_1_1_Sycl_PVC_MKL-tpetra_sacado_ifpack2/41/changes#detail))
```

The Sacado fix in #14612 would be unrelated to the error type, I suspect some of the recent Teuchos changes based on the linker error? I'll need to bisect to determine


### Steps to Reproduce
1. SHA1: 78dc00aa6bf2cb486f224786817e96e4a6ef07c2
2. Configure + build notes: (blake testbed, PV queue):
```bash

ssh blake
salloc -N 1 -p PV

source /projects/x86-64-icelake-rocky8/spack-config/blake-setup-user-module-env.sh
module purge
module load cmake git gcc/12.2.0 intel-oneapi-compilers/2025.1.1 intel-oneapi-mkl/2025.1.0 intel-oneapi-tbb/2022.2.0 intel-oneapi-dpl/2022.9.0 intel-oneapi-mpi/2021.16.0


export TRILINOS_DIR=<path-to-source>

cmake \
  -B build \
  -D CMAKE_CXX_COMPILER="$(which icpx)" \
  -D CMAKE_C_COMPILER="$(which icx)" \
  -D CMAKE_Fortran_COMPILER="$(which ifx)" \
  -D CMAKE_CXX_FLAGS="-fp-model=precise --gcc-toolchain=${GCC_ROOT} -Wno-deprecated-declarations " \
  -D CMAKE_EXE_LINKER_FLAGS="-flink-huge-device-code" \
  -D CMAKE_CXX_STANDARD=20 \
  -D BUILD_SHARED_LIBS=ON \
  -DTPL_ENABLE_MPI=OFF \
  -DTPL_ENABLE_BLAS:BOOL=ON \
    -D BLAS_LIBRARY_DIRS:FILEPATH="${MKLROOT}/lib/intel64" \
    -D BLAS_LIBRARY_NAMES="mkl_rt" \
  -DTPL_ENABLE_LAPACK:BOOL=ON \
    -D LAPACK_LIBRARY_DIRS:FILEPATH="${MKLROOT}/lib/intel64" \
    -D LAPACK_LIBRARY_NAMES="mkl_rt" \
  -DTrilinos_ENABLE_ALL_PACKAGES=OFF \
  -DTrilinos_ENABLE_ALL_OPTIONAL_PACKAGES=OFF \
  -DTrilinos_ENABLE_TESTS=ON \
  -DTrilinos_MUST_FIND_ALL_TPL_LIBS=TRUE \
  -DTrilinos_ENABLE_OpenMP=OFF \
  -DTrilinos_ENABLE_Kokkos=ON \
  -D Kokkos_ARCH_INTEL_PVC=ON \
  -D Kokkos_ENABLE_SYCL=ON \
   -D Kokkos_ENABLE_TESTS=OFF \
   -D Kokkos_ENABLE_ONEDPL=OFF \
   -D Kokkos_ENABLE_IMPL_VIEW_LEGACY=ON \
  -DTrilinos_ENABLE_KokkosKernels=ON \
   -D KokkosKernels_ENABLE_TESTS=OFF \
  -DTrilinos_ENABLE_Tpetra=ON \
   -D Tpetra_ENABLE_TESTS=ON \
   -D Tpetra_INST_SYCL=ON \
   -D Tpetra_INST_SERIAL=ON \
  -DTrilinos_ENABLE_Sacado=ON \
   -D Sacado_ENABLE_TESTS=ON \
   -D Sacado_GTestSuite_DISABLE=ON \
  -DTrilinos_ENABLE_Ifpack2=ON \
   -D Ifpack2_ENABLE_TESTS=ON \
\
  -DTPL_ENABLE_Matio=OFF \
\
$TRILINOS_DIR

cmake --build --parallel 16
```

--- a/packages/teuchos/core/src/Teuchos_EnvVariables.cpp
+++ b/packages/teuchos/core/src/Teuchos_EnvVariables.cpp
@@ -190,6 +190,8 @@ T idempotentlyGetEnvironmentVariable(
 }
 
 
+template std::string getEnvironmentVariable<std::string>(std::string_view, const std::string);
+
 template std::string idempotentlyGetEnvironmentVariable<std::string>(std::string&, bool&, const std::string_view, const std::string);
 template int idempotentlyGetEnvironmentVariable<int>(int&, bool&, const std::string_view, const int);
 template unsigned long idempotentlyGetEnvironmentVariable<unsigned long>(unsigned long&, bool&, const std::string_view, const unsigned long);

